VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -148,4 +123,90 @@ public void onClickAddComment(@NonNull DiscussionComment comment) {
+
+        private GetCommentListTask getCommentListTask;
+        private GetCommentListTask getEndorsedListTask;
+
+        private ThreadComments unendorsedComments;
+        private int nextPage = 1;
+        private boolean isQuestionTypeThread = false;
+        private boolean isEndorsedFetched = false;
+
+        public CommentsLoader(boolean isQuestionTypeThread) {
+            this.isQuestionTypeThread = isQuestionTypeThread;
+        }
+
+        public void loadComments(@NonNull InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback) {
+            this.callback = callback;
+            if (isQuestionTypeThread && !isEndorsedFetched) getEndorsedList();
+            getCommentList();
+        }
+
+        /**
+         * Gets the list of endorsed answers for a {@link DiscussionThread.ThreadType#QUESTION}
+         * type discussion thread
+         */
+        protected void getEndorsedList() {
+            if (getEndorsedListTask != null) {
+                getEndorsedListTask.cancel(true);
+            }
+            getEndorsedListTask = new GetCommentListTask(getActivity(),
+                    discussionThread.getIdentifier(), 1, isQuestionTypeThread, true) {
+                @Override
+                public void onSuccess(ThreadComments threadComments) {
+                    isEndorsedFetched = true;
+                    courseDiscussionResponsesAdapter.addToTop(threadComments.getResults());
+                    // If the unendorsed call returned earlier than this one
+                    if (unendorsedComments != null) finalizeCommentLoading();
+                }
+
+                @Override
+                public void onException(Exception ex) {
+                    logger.error(ex);
+                }
+            };
+            getEndorsedListTask.setProgressCallback(null);
+            getEndorsedListTask.execute();
+        }
+
+        protected void getCommentList() {
+            if (getCommentListTask != null) {
+                getCommentListTask.cancel(true);
+            }
+            getCommentListTask = new GetCommentListTask(getActivity(),
+                    discussionThread.getIdentifier(),
+                    nextPage, isQuestionTypeThread, false) {
+                @Override
+                public void onSuccess(final ThreadComments threadComments) {
+                    unendorsedComments = threadComments;
+                    if (isQuestionTypeThread) {
+                        if (isEndorsedFetched) finalizeCommentLoading();
+                    } else {
+                        finalizeCommentLoading();
+                    }
+                }
+
+                @Override
+                public void onException(Exception ex) {
+                    logger.error(ex);
+                }
+            };
+            getCommentListTask.setProgressCallback(null);
+            getCommentListTask.execute();
+        }
+
+        private void finalizeCommentLoading() {
```

Comment:
It's less descriptive.

On Wed, Dec 16, 2015 at 4:48 PM, Mi@N Kh@LiD <notifications@github.com>
wrote:

> In
> VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java
> <https://github.com/edx/edx-app-android/pull/475#discussion_r47766435>:
>
> > +                        if (isEndorsedFetched) finalizeCommentLoading();
> > +                    } else {
> > +                        finalizeCommentLoading();
> > +                    }
> > +                }
> > +
> > +                @Override
> > +                public void onException(Exception ex) {
> > +                    logger.error(ex);
> > +                }
> > +            };
> > +            getCommentListTask.setProgressCallback(null);
> > +            getCommentListTask.execute();
> > +        }
> > +
> > +        private void finalizeCommentLoading() {
>
> whats the problem with current name ? [image: :open_mouth:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/475/files#r47766435>.
>